data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/impl/http/WebClientCloudEventSender.java

```diff
@@ -176,7 +176,9 @@ private Future<?> send(final CloudEvent event, final Promise<HttpResponse<Buffer
     return VertxMessageFactory
       .createWriter(client.postAbs(target)
         .timeout(this.consumerVerticleContext.getEgressConfig().getTimeout() <= 0 ? DEFAULT_TIMEOUT_MS : this.consumerVerticleContext.getEgressConfig().getTimeout())
-        .putHeader("Prefer", "reply"))
+        .putHeader("Prefer", "reply")
+        .putHeader("Kn-Namespace", this.consumerVerticleContext.getEgress().getReference().getNamespace())
```

Comment:
so we set it all the time.

but benefit from it on the "mesh case", since we can use that for filtering/tweaking ?

Reply:
Yes